core/src/test/java/io/questdb/test/griffin/AlterTableChangeColumnTypeTest.java

```diff
@@ -0,0 +1,971 @@
+            engine.releaseInactive();
+
+            insert("insert into x(c, timestamp) values('def', now())", sqlExecutionContext);
+            assertSql("c\ndef\n", "select c from x limit -1");
+        });
+    }
+
+    @Test
+    public void testConvertFromSymbolToStringDedupFlagNotAllowed() throws Exception {
+        assumeWal();
+        assertMemoryLeak(() -> {
+            createX();
+
+            ddl("alter table x dedup enable upsert keys(timestamp, ik)");
+            drainWalQueue();
+            checkDedupSet("x", "ik", true);
+
+            try {
+                ddl("alter table x alter column ik type varchar");
+                Assert.fail();
+            } catch (SqlException ex) {
+                TestUtils.assertContains(ex.getFlyweightMessage(), "cannot change type of deduplicated key column 'ik' to variable size type 'VARCHAR', deduplication is only supported for fixed size types");
+                Assert.assertEquals(35, ex.getPosition());
+            }
+
+            ddl("alter table x dedup disable");
+            drainWalQueue();
+
+            // In one go, enable dedup and change type
+            ddl("alter table x dedup enable upsert keys(timestamp, ik)");
+            ddl("alter table x alter column ik type varchar");
+            drainWalQueue();
+
+            checkDedupSet("x", "ik", false);
+
+            engine.releaseInactive();
+            checkDedupSet("x", "ik", false);
+
+            insert("insert into x(ik, d, timestamp) values('abc', 2, '2044-02-24')", sqlExecutionContext);
+            insert("insert into x(ik, d, timestamp) values('abc', 3, '2044-02-25')", sqlExecutionContext);
+            insert("insert into x(ik, d, timestamp) values('def', 4, '2044-02-25')", sqlExecutionContext);
+
+            drainWalQueue();
+
+            assertSql("timestamp\td\tik\n" +
+                    "2018-01-01T02:00:00.000000Z\t0.04488373772232379\tCPSW\n" +
+                    "2044-02-24T00:00:00.000000Z\t2.0\tabc\n" +
+                    "2044-02-25T00:00:00.000000Z\t4.0\tdef\n", "select timestamp, d, ik from x limit -3");
+        });
+    }
+
+    @Test
+    public void testConvertInvalidColumnFailsWriterIsOk() throws Exception {
+        assumeNonWal();
+        assertMemoryLeak(() -> {
+            createX();
+
+            try (TableWriter writer = getWriter("x")) {
+                writer.changeColumnType("non_existing", ColumnType.INT, 0, false, false, 0, false, null);
+                Assert.fail();
+            } catch (CairoException e) {
+                TestUtils.assertContains(e.getFlyweightMessage(), "cannot change column type, column does not exists");
+            }
+
+            insert("insert into x(c, timestamp) values('abc', now())", sqlExecutionContext);
+            assertSql("c\nabc\n", "select c from x limit -1");
+        });
+    }
+
+    @Test
+    public void testFixedSizeColumnEquivalentToCast() throws Exception {
+        assertMemoryLeak(() -> {
+            assumeNonWal();
```

Comment:
why just non-WAL?
there are other non-WAL only tests in this class too.
would be good to run these for WAL tables too.

Reply:
Non-Wal is slightly faster, these tests don't test any WAL specifics, simply the conversion correctness.
We have a growing test execution time, I'm trying to be leaner here running necessary tests cases only.

Reply:
I think you are right that we do not need to run the tests for both, but I would rather run it for WAL tables where it is possible. Not a big deal though, will resolve the conversation.